app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/home/HomeScreen.kt

```diff
@@ -53,45 +67,61 @@ fun HomeScreen(viewModel: HomeViewModel = hiltViewModel()) {
 
     val surveyList by viewModel.surveyList.collectAsState()
     val error by viewModel.error.collectAsState()
-    var selectedSurveyNumber by remember { mutableStateOf(0) }
-    val swipeableState = rememberSwipeableState(initialValue = "M")
-    val sizePx = LocalDensity.current.run { LocalConfiguration.current.screenWidthDp.toDp().toPx() }
+    var selectedSurveyNumber by remember { mutableStateOf(START_SURVEY_NUMBER) }
+    val swipeableState = rememberSwipeableState(initialValue = MID_STATE)
+    val endAnchor = LocalDensity.current.run { LocalConfiguration.current.screenWidthDp.toDp().toPx() }
     // Maps anchor points (in px) to states
-    val anchors = mapOf(0f to "L", sizePx / 2 to "M", sizePx to "R")
+    val anchors = mapOf(START_ANCHOR to LEFT_STATE, endAnchor / 2 to MID_STATE, endAnchor to RIGHT_STATE)
     var threshold by remember { mutableStateOf(IDLE) }
     val isRefreshing by viewModel.isRefreshing.collectAsState()
+    val userAvatar by viewModel.userAvatar.collectAsState()
+
+    val shimmerDescription = stringResource(id = R.string.home_shimmer)
+    val surveyContentDescription = stringResource(id = R.string.home_survey_content)
 
     SwipeRefresh(
         state = rememberSwipeRefreshState(isRefreshing),
         onRefresh = {
-            // TODO: Implement swipe refresh here
+            viewModel.clearCacheAndFetch()
+            selectedSurveyNumber = START_SURVEY_NUMBER
         },
         modifier = Modifier.fillMaxSize()
     ) {
         LazyColumn {
             if (isRefreshing && surveyList.isEmpty()) {
-                item { HomeScreenShimmerLoading(modifier = Modifier.fillParentMaxSize()) }
+                item {
+                    HomeScreenShimmerLoading(
+                        modifier = Modifier
+                            .fillParentMaxSize()
+                            .semantics { contentDescription = shimmerDescription }
+                    )
+                }
```

Comment:
Let's revert this change and put the `contentDescription` directly into `ShimmerHomeScreen`
```suggestion
                item { HomeScreenShimmerLoading(modifier = Modifier.fillParentMaxSize()) }
```
And in `ShimmerHomeScreen`:
```kotlin
@Composable
fun ShimmerHomeScreen(brush: Brush, modifier: Modifier) {
    val shimmerDescription = stringResource(id = R.string.home_shimmer)
    ConstraintLayout(
        modifier = modifier
            .background(Color.Black)
            .semantics { contentDescription = shimmerDescription }
    ) {